core/data/tests/anonymous_struct_with_rename/output.py

```diff
@@ -0,0 +1,57 @@
+"""
+ Generated by typeshare 1.12.0
+"""
+from __future__ import annotations
+
+from enum import Enum
+from pydantic import BaseModel, ConfigDict, Field
+from typing import List, Union
+
+
+class AnonymousStructWithRenameListInner(BaseModel):
+    """
+    Generated type representing the anonymous struct variant `List` of the `AnonymousStructWithRename` Rust enum
+    """
+    list: List[str]
+
+
+class AnonymousStructWithRenameLongFieldNamesInner(BaseModel):
+    """
+    Generated type representing the anonymous struct variant `LongFieldNames` of the `AnonymousStructWithRename` Rust enum
+    """
+    model_config = ConfigDict(populate_by_name=True)
+
+    some_long_field_name: str
+    and_: bool = Field(alias="and")
+    but_one_more: List[str]
+
+
+class AnonymousStructWithRenameKebabCaseInner(BaseModel):
+    """
+    Generated type representing the anonymous struct variant `KebabCase` of the `AnonymousStructWithRename` Rust enum
+    """
+    model_config = ConfigDict(populate_by_name=True)
+
+    another_list: List[str] = Field(alias="another-list")
+    camel_case_string_field: str = Field(alias="camelCaseStringField")
+    something_else: bool = Field(alias="something-else")
+
+
+class AnonymousStructWithRenameTypes(str, Enum):
+    LIST = "list"
+    LONG_FIELD_NAMES = "longFieldNames"
+    KEBAB_CASE = "kebabCase"
+
+class AnonymousStructWithRenameList(BaseModel):
+    type: AnonymousStructWithRenameTypes = AnonymousStructWithRenameTypes.LIST
```

Comment:
It's possible I'm misunderstanding pydantic, but shouldn't this be a `Literal` parameter? I had understood this `=` to just set a default value. 

https://docs.pydantic.dev/2.0/usage/types/unions/

Reply:
Fixed